controllers/articles.controllers.js

```diff
@@ -9,7 +9,12 @@ exports.getArticlesById = (req, res, next) => {
 };
 
 exports.getArticles = (req, res, next) => {
-    fetchArticles().then((articles) => {
+    const { sort_by: userQuery, order: userOrder, topic: userTopic } = req.query;
+        fetchTopics()
```

Comment:
You could utilise a promise.all to check if the topic exists rather than nesting .then's within other .thens